src/common/mfu_flist_remove.c

```diff
@@ -588,6 +589,81 @@ void mfu_flist_unlink(mfu_flist flist)
     int levels, minlevel;
     mfu_flist* lists;
     mfu_flist_array_by_depth(flist, &levels, &minlevel, &lists);
+    mfu_flist pstatlist;
+    uint64_t size = mfu_flist_size(flist);
+    char* strings[size];
+
+    /* if traceless, dump the stat of each item's pdir */
+    if (traceless) {
+        uint64_t idx;
+
+        for (idx = 0; idx < size; idx++) {
+            /* stat the item */
+            struct stat st;
+            /* get name of item */
+            const char* name = mfu_flist_file_get_name(flist, idx);
+            char *pdir = MFU_STRDUP(name);
+
+            dirname(pdir);
+
+            int len = strlen(pdir);
+
+            strings[idx] = (char *) MFU_MALLOC(len + 1);
+
+            strncpy(strings[idx], pdir, len);
+
+            mfu_free(&pdir);
+        }
+
+        pstatlist = mfu_flist_new();
+
+        mfu_flist_set_detail(pstatlist, 1);
+
+        /* allocate arrays to hold result from DTCMP_Rankv_strings call to
+        * assign group and rank values to each item */
+        uint64_t output_bytes = size * sizeof(uint64_t);
+        uint64_t groups;
+        uint64_t* group_ids   = (uint64_t*) MFU_MALLOC(output_bytes);
+        uint64_t* group_ranks = (uint64_t*) MFU_MALLOC(output_bytes);
+        uint64_t* group_rank  = (uint64_t*) MFU_MALLOC(output_bytes);
+
+        DTCMP_Rankv_strings(size, strings, &groups, group_ids, group_ranks,
+                           group_rank, DTCMP_FLAG_NONE, MPI_COMM_WORLD);
+
+        for (idx = 0; idx < size; idx++) {
+            if(group_rank[idx] == 0) {
+                /* stat the item */
+                struct stat st;
+                char *pdir = strings[idx];
+                int status = mfu_lstat(pdir, &st);
+
+                if (status != 0) {
+                    MFU_LOG(MFU_LOG_ERR, "mfu_lstat(): %d", status);
+                    mfu_free(&pdir);
```

Comment:
Need to do this?  I think it would lead to a possible double free at line 652 where we free each value in the strings array.